pkg/daemon/drain.go

```diff
@@ -176,6 +176,20 @@ func isDrainRequired(actions, diffFileSet []string, readOldFile, readNewFile Rea
 	return true, nil
 }
 
+func (dn *Daemon) drainIfRequired(actions, diffFileSet []string, readOldFile, readNewFile ReadFileFunc) error {
```

Comment:
Can you explain what this is doing? Because I don't quite understand or know if this refactor should be pulled into master without accompanying code to make sense of it?

Reply:
It's just factoring code out of `update()` so it doesn't have to be duplicated when we want to perform drains for a layered update

Reply:
I can't help but feel like that's should probably live with the other code that needs it? Feels weird to merge it if it's needed for something that isn't in master  or a clear optimization for master

Reply:
The counterbalance here is avoiding a later merge of layering into master being too large and painful, as well as the ongoing cost of rebases.

It's highly likely that there will be an intersection between layering and hypershift work for example.  And having prep patches merged into master helps avoid conflicts there too.

In this particular case, I think

- The code isn't *significantly* better or anything, but IMO it is a bit better
- The code isn't difficult to review
- CI covers all the functionality here

IMO, that on balance argues for a master merge.